### PR TITLE
Manage group memberships via RBAC

### LINKED ISSUE #17: Manage User Groups
## Context
In #1 we implemented LDAP group sync. Since we'll move away from LDAP as authentication provider (to OIDC) we'll need another way to manage the groups.
Implement the feature in this component to manage user groups:
It should be possible to assign a list of usernames to a certain group. For example all VSHN users can be mapped to the "VSHN openshiftroot" group that way. Additionally this allows also customers to introduce groups of their own users.

## Alternatives
Using LDAP group sync is currently the only option to manage user groups in OpenShift.

## Acceptance Criteria

* Given a hierarchy, when defining the group memberships in the tenant level (or higher up), then the RBAC groups are created for all clusters that are part of the hierarchy.
* Given a defined group memberships in a hierarchy, when a specific user is removed/nullified from a group on a more specific level (e.g. cluster), then the specific user is not in the RBAC group.
* Given a defined group memberships in a hierarchy, when an additional user is added to a group, then the RBAC group includes the given user.
* Given a defined group memberships in a hierarchy, when the group is removed/nullified in lower level of the hierarchy, then the RBAC group will not be created

### PR DESCRIPTION
**Pull request by ccremer**
* Adds possibility to manage group memberships for users.

TL;DR: Interface looks like this:
```yaml
parameters:
  openshift4_authentication:
    groupMemberships:
      test-group:
        users:
          test-username: {}
          absent-username:
            state: absent
      absent-group:
        state: absent
        users:
          irrelevant-user: {}
      empty-group:
        users: {}
```

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Link this PR to related issues.

### CODE REVIEW

**Comment by ccremer** [comment]
I actually tried to set up unit tests with Go. However, I was not able to find the API spec for Users or Groups for `rbac.authorization.k8s.io`. I couldn't find any references to `Group` struct which would be parsed from the output YAML. The RBAC v1 spec in https://github.com/kubernetes/api/blob/master/rbac/v1/types.go doesn't provide it.

I'll try re-creating our own types.

**Comment by ccremer** [comment]
I tried by creating our own type. Unfortunately not so easy to include in the kubernetes schemes. I propose to leave the unit test for later. At least it's testable manually by inspecting the output yaml when playing around in `tests/defaults.yml`

**Review by simu — APPROVED** [review]
Looks good for now. IMO the inventory structure provides a good base to expand on later if there's additional requirements.
